pywbem_mock/_mainprovider.py

```diff
@@ -401,17 +387,21 @@ def _iter_association_classes(self, namespace):
     @staticmethod
     def _find_instance(iname, instance_store, copy_inst=None):
         """
-        Find an instance in the instance repo by iname and return the
-        index of that instance.
+        Find an instance in the repository by iname and return
+        that instance. the copy_inst controls whether the original
+        instance in the object store is returned on only a copy.  The
+        only time the original should be returned is when the user is
+        certain that the returned object WILL NOT be modified.
 
         Parameters:
 
           iname: CIMInstancename to find
 
-          instance_store: the instance repo to search
+          instance_store (:class:`~pywbem_mock.BaseObjectStore):
+            The object store to search for the instance
```

Comment:
The `MainProvider` object knows its CIM repository, and that repository has only one instance object store. I suggest to remove the `instance_store` parameter and look the instance object store up from the repository on `self`.

Reply:
I was very inconsistent in the use of namespace vs. the get_...store variables in the local methods.  My intention was to use the get_..._store as much as possible because that is a clear indication that the namespace has already been validated but that sort of fell apart because a number of these methods use namespace as an exception component.   I think the goal should be to use the get..._store methods as much as possible except when the namespace name itself is required in the called method. In any case, not logical to have both namespace and get_..._store in the call.

I agree that we can drop the combination of 

validate_namespace
...store =get_..._store(namespace)

since both call the repository to get the namespace. which is pure duplication.

**IN_PROCESS**

Reply:
How about removing the `validate_namespace()` call from all `get_..._store()` methods and calling `validate_namespace()` explicitly, either on the passed `namespace` argument or if that does not exist on the namespace of the target object?

Reply:
This has been DONE per another comment

**DONE**